File: Models/Desire.swift
import SwiftData

@Model
final class Desire {
    var id: UUID
    var cropName: String
    var variety: String?
    var isFulfilled: Bool
    var isExpired: Bool
    var isCancelled: Bool
    var notes: String?
    var createdAt: Date
    var updatedAt: Date

    @Relationship(deleteRule: .cascade, inverse: \Garden.desires)
    var garden: Garden?

    @Relationship(deleteRule: .nullify, inverse: \PlannedPlanting.desire)
    var plannedPlantings: [PlannedPlanting] = []

    init(
        cropName: String,
        variety: String? = nil,
        notes: String? = nil,
        garden: Garden? = nil
    ) {
        self.id = UUID()
        self.cropName = cropName
        self.variety = variety
        self.isFulfilled = false
        self.isExpired = false
        self.isCancelled = false
        self.notes = notes
        self.garden = garden
        self.createdAt = Date()
        self.updatedAt = Date()
    }

    var displayName: String {
        if let variety = variety, !variety.isEmpty {
            return "\(variety) \(cropName)"
        }
        return cropName
    }
}

File: shared/src/commonMain/kotlin/com/soilandsupper/shared/domain/model/Desire.kt
package com.soilandsupper.shared.domain.model

data class Desire(
    val id: Long = 0,
    val cropName: String,
    val variety: String? = null,
    val notes: String? = null,
    val isFulfilled: Boolean = false,
    val isCancelled: Boolean = false,
    val isExpired: Boolean = false,
    val gardenId: Long? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
) {
    val displayName: String
        get() = if (!variety.isNullOrBlank()) "$variety $cropName" else cropName
}

